common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -2205,9 +2205,8 @@ public static enum ConfVars {
         "padding tolerance config (hive.exec.orc.block.padding.tolerance)."),
     HIVE_ORC_CODEC_POOL("hive.use.orc.codec.pool", false,
         "Whether to use codec pool in ORC. Disable if there are bugs with codec reuse."),
-    HIVE_USE_STATS_FROM("hive.use.stats.from","iceberg","Use stats from iceberg table snapshot for query " +
-        "planning. This has three values metastore, puffin and iceberg"),
-
+    HIVE_ICEBERG_STATS_SOURCE("hive.iceberg.stats.source","iceberg","Use stats from iceberg " +
```

Comment:
Is it worth adding a PatternSet as it is used for `HIVESTATSDBCLASS("hive.stats.dbclass", "fs", new PatternSet("custom", "fs"),`? 

In that way, it would be easy to understand what are the accepted values, for the first look at the code. 

Also, I wonder, if is there any other way to do similar stuff with settings?